packages/legacy/core/App/components/listItems/ContactListItem.tsx

```diff
@@ -1,24 +1,47 @@
-import type { ConnectionRecord } from '@aries-framework/core'
+import type {
+  BasicMessageRecord,
+  ConnectionRecord,
+  CredentialExchangeRecord,
+  ProofExchangeRecord,
+} from '@aries-framework/core'
 
+import { useBasicMessagesByConnectionId } from '@aries-framework/react-hooks'
 import { StackNavigationProp } from '@react-navigation/stack'
-import React, { useCallback, useMemo } from 'react'
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
 import { useTranslation } from 'react-i18next'
-import { View, StyleSheet, TouchableOpacity, Image } from 'react-native'
+import { View, StyleSheet, TouchableOpacity, Image, Text } from 'react-native'
 
 import { useTheme } from '../../contexts/theme'
+import { useCredentialsByConnectionId } from '../../hooks/credentials'
+import { useProofsByConnectionId } from '../../hooks/proofs'
+import { Role } from '../../types/chat'
 import { ContactStackParams, Screens, Stacks } from '../../types/navigators'
-import { formatTime } from '../../utils/helpers'
+import {
+  formatTime,
+  getCredentialEventLabel,
+  getCredentialEventRole,
+  getProofEventLabel,
+  getProofEventRole,
+} from '../../utils/helpers'
 import { testIdWithKey } from '../../utils/testable'
-import Text from '../texts/Text'
 
+interface CondensedMessage {
+  text: string
+  createdAt: Date
+}
 interface Props {
   contact: ConnectionRecord
   navigation: StackNavigationProp<ContactStackParams, Screens.Contacts>
 }
 
 const ContactListItem: React.FC<Props> = ({ contact, navigation }) => {
   const { t } = useTranslation()
-  const { TextTheme, ColorPallet } = useTheme()
+  const { TextTheme, ColorPallet, ListItems } = useTheme()
+  const connectionId = contact.id
```

Comment:
I don't think `connectionId` is used anywhere besides the 3 following lines. Why not just `useBasicMessagesByConnectionId(contact.id)`

Reply:
done